src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -150,6 +157,59 @@ public async Task<IActionResult> Validate()
         }
     }
 
+    [Route("PaymentRequest/{orderId}")]
+    public async Task<IActionResult> PaymentRequest(string orderId)
+    {
+        if (await _contentManager.GetAsync(orderId) is not { } order) return NotFound();
+
+        // Users should only see their own Orders.
+        if (order.Author != User.Identity.Name)
+        {
+            return NotFound();
+        }
+
+        var orderPart = order.As<OrderPart>();
+
+        // If status is not Pending or there are no line items, there is nothing to be done.
+        if (!string.Equals(orderPart.Status.Text, OrderStatuses.Pending, StringComparison.OrdinalIgnoreCase) ||
+            !orderPart.LineItems.Any())
+        {
+            return NotFound();
+        }
+
+        var currency = orderPart.LineItems[0].LinePrice.Currency;
+        var singleCurrencyTotal = new Amount(0, currency);
+        foreach (var item in orderPart.LineItems)
+        {
+            singleCurrencyTotal += item.LinePrice;
+        }
+
+        if (singleCurrencyTotal.Value <= 0)
+        {
+            return NotFound();
+        }
+
+        var stripeApiSettings = (await _siteService.GetSiteSettingsAsync()).As<StripeApiSettings>();
+        var paymentAmount = _stripePaymentService.GetPaymentAmount(singleCurrencyTotal.Value, currency.CurrencyIsoCode);
+        var paymentIntent = await _stripePaymentService.CreatePaymentIntentAsync(paymentAmount, singleCurrencyTotal);
+
+        _session.Save(new OrderPayment
+        {
+            OrderId = order.ContentItemId,
+            PaymentIntentId = paymentIntent.Id,
+        });
+
+        return View(new
+        {
+            SingleCurrencyTotal = singleCurrencyTotal,
+            NetTotal = singleCurrencyTotal,
+            GrossTotal = new Amount(0, currency),
```

Comment:
This feels like an unnecessary workaround. You could just fix the line 38 in StripeCheckout.cshtml so it still works if the GrossTotal prop is not passed instead of passing a dummy value.